frontend/src/metabase/search/components/filters/TypeFilter/TypeFilter.tsx

```diff
@@ -2,10 +2,21 @@ import { t } from "ttag";
 import type { SearchSidebarFilterComponent } from "metabase/search/types";
 import { TypeFilterContent } from "metabase/search/components/filters/TypeFilter/TypeFilterContent";
 import { TypeFilterDisplay } from "metabase/search/components/filters/TypeFilter/TypeFilterDisplay";
+import type { EnabledSearchModelType } from "metabase-types/api";
+import { isEnabledSearchModelType } from "metabase/search/utils/enabled-search-type/enabled-search-type";
 
 export const TypeFilter: SearchSidebarFilterComponent<"type"> = {
   iconName: "dashboard",
   title: t`Content type`,
   DisplayComponent: TypeFilterDisplay,
   ContentComponent: TypeFilterContent,
+  fromUrl: value => {
+    if (Array.isArray(value)) {
+      return value.filter((v): v is EnabledSearchModelType =>
+        isEnabledSearchModelType(v),
+      );
```

Comment:
```suggestion
      return value.filter(isEnabledSearchModelType);
```

Reply:
The only reason I don't do this is because the `filter` function is technically
```
Array.filter((element, index, array) => {...whatever...}) 
```
- doing it this way makes the code more readable and explicit about the parameters that we're using, and we should probably do this in our other array functions

Here's the article that changed my views on this: https://jakearchibald.com/2021/function-callback-risks/

Reply:
Fair enough :+1: